File: backend/app/Models/Booking.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Booking extends Model
{
    protected $fillable = [
        'tenant_id',
        'apartment_id',
        'start_date',
        'end_date',
        'location',
        'status',
        'modification_status',
        'payment_method',
        'payment_status',
    ];

    public function tenant()
    {
        return $this->belongsTo(User::class, 'tenant_id');
    }

    public function apartment()
    {
        return $this->belongsTo(Apartment::class, 'apartment_id');
    }
}

File: frontend/main/lib/pagesowner/models/booking.dart
class Booking {
  final String id;
  final String propertyId;
  final DateTime startDate;
  final DateTime endDate;
  final String status;

  Booking({
    required this.id,
    required this.propertyId,
    required this.startDate,
    required this.endDate,
    required this.status,
  });
}
